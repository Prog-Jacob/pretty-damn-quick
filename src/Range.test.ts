import { Range } from "./Range";

describe("Range", () => {
  it("should instantiate with correct parameters", () => {
    const range = new Range(2, 5);
    expect(range).toBeInstanceOf(Range);
  });

  it("should throw TypeError when parameters are flipped or equal", () => {
    expect(() => new Range(5, 2)).toThrowErrorMatchingSnapshot();
    expect(() => new Range(3, 3)).toThrow(TypeError);
  });

  it("rangeStart returns inclusiveLowerBound - 1", () => {
    const range = new Range(2, 5);
    expect(range.rangeStart()).toBe(1);
  });

  it("rangeEnd returns exclusiveUpperBound - 1", () => {
    const range = new Range(2, 5);
    expect(range.rangeEnd()).toBe(4);
  });

  it("isWithinRange returns true for values in range", () => {
    const range = new Range(2, 5);
    expect(range.isWithinRange(2)).toBe(true);
    expect(range.isWithinRange(4)).toBe(true);
  });

  it("isWithinRange returns false for values out of range", () => {
    const range = new Range(2, 5);
    expect(range.isWithinRange(1)).toBe(false);
    expect(range.isWithinRange(5)).toBe(false);
  });
});
