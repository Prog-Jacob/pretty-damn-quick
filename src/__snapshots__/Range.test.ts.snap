// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should throw TypeError when parameters are flipped or equal 1`] = `"inclusiveLowerBound must be strictly less than exclusiveUpperBound: 5 >= 2"`;
